src/main/java/io/github/battlepass/quests/service/executor/QuestExecutionBuilderImpl.java

```diff
@@ -0,0 +1,120 @@
+package io.github.battlepass.quests.service.executor;
+
+import io.github.battlepass.objects.quests.variable.ExecutableQuestResult;
+import io.github.battlepass.objects.quests.variable.QuestResult;
+import io.github.battlepass.quests.workers.pipeline.QuestPipeline;
+import io.github.battlepass.service.CheckHelper;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import java.math.BigInteger;
+import java.util.function.UnaryOperator;
+
+public class QuestExecutionBuilderImpl implements QuestExecutionBuilder {
+    private final QuestPipeline pipeline;
+    private Player player;
+
+    private final String questType;
+    private BigInteger progress;
+    private boolean overrideUpdate;
+
+    private ExecutableQuestResult questResult;
+
+    public QuestExecutionBuilderImpl(QuestPipeline pipeline, String questType) {
+        this.pipeline = pipeline;
+        this.questType = questType;
+    }
+
+    @Override
+    public void buildAndExecute() {
+        QuestExecution execution = this.build();
+        this.pipeline.handle(execution);
+    }
+
+    @Override
+    public QuestExecution build() {
+        String baseMessage = "QuestExecution Build -> %s must be set.";
+        if (this.player == null)
+            throw new IllegalStateException(String.format(baseMessage, "Player"));
+        if (this.questType == null || this.questType.isEmpty())
+            throw new IllegalStateException(String.format(baseMessage, "Quest type"));
+        if (this.progress == null)
+            throw new IllegalStateException(String.format(baseMessage, "Progress"));
+        if (this.questResult == null)
+            this.questResult = new ExecutableQuestResult().root("none");
+        return new QuestExecution(this.player, this.questType, this.progress, this.overrideUpdate, this.questResult);
+    }
+
+    @Override
+    public QuestExecutionBuilder player(Player player) {
+        CheckHelper.notNull(player, "Quest execution player");
+        this.player = player;
+        return this;
+    }
+
+    @Deprecated
+    @Override
+    public QuestExecutionBuilder questResult(UnaryOperator<QuestResult> resultOperator) {
+        this.questResult = (ExecutableQuestResult) resultOperator.apply(new ExecutableQuestResult());
+        return this;
+    }
+
+    @Override
+    public QuestExecutionBuilder root(String root) {
+        if (this.questResult == null)
+            this.questResult = new ExecutableQuestResult();
```

Comment:
Please wrap this in {}, and all other instances of this. Once again personal preference, but just makes it more robust for future, and looks more structured in my opinion.